File: S_3/Programação Orientada a Objetos/Exercicio5.java
import java.io.IOException;

public class Exercicio5 {
    public static boolean primo(int n) {
        boolean p = true;
        int t = (int) (Math.sqrt(n));
        for (int i = 2; i <= t; ++i) {
            if (Math.floorMod(n, i) == 0) {
                p = false;
                break;
            }
        }
        return p;
    }

    public static void main(String[] args) throws IOException {
        System.out.println("Digite um número:");
        int n = EntradaTeclado.leInt();
        if (primo(n)) {
            System.out.println("É primo.");
        } else {
            while (!primo(n)) {
                n--;
            }
            System.out.println("O primeiro número primo menor é "+n);
        }

    }
}

File: S_3/Programação Orientada a Objetos/Exercicio5.py
import math
def primo (n):
    p = True
    t = int(math.pow(n,0.5))
    i = 2
    while i <= t:
        if n % i == 0:
            p = False
            break
        i+=1
    return p
n = int(input("Digite um inteiro:"))
if(primo(n)):
    print("É primo.")
else:
    while not primo(n):
        n-=1
    print("O primeiro número primo menor é "+str(n))

